### ISSUE TITLE
llama2-13b-tp8 long sentence input error.[BUG]

### ISSUE THREAD

**Issue by zhyajie** [microsoft/DeepSpeed#4855]
path = "path/to/llama2_13b_huggingface_path"
pipe = mii.pipeline(path)
prompt = "test "*6000
response = self.pipe(prompt, max_new_tokens=512, max_length=133120, do_sample=False)

python3.9/site-packages/deepspeed/inference/v2/ragged/sequence_descriptor.py", line 258, in extend_kv_cache
shadow_alloc_group[cur_blocks:cur_blocks + new_blocks].copy_(new_group_ids)
RuntimeError:  The size of tensor a (8) must match the size of tensor b (12) at non-singleton dimension 0
I gebug the code find:
len(shadow_alloc_group) is 32 
len(new_group_ids) is 12
cur_blocks  = 24
new_blocks = 12

**Comment by mrwyattii** [comment]
Hi @zhyajie I'm wondering if this is a problem with the prompt length. If you run with just a prompt of `"test"` does that succeed?

**Comment by zhyajie** [comment]
> Hi @zhyajie I'm wondering if this is a problem with the prompt length. If you run with just a prompt of `"test"` does that succeed?

Yes, I found that when the prompt length is less than 8192, the inference process is OK, but when it exceeds 8192, the above error will occur.
